src/client/metric.rs

```diff
@@ -286,11 +291,170 @@ impl<T: MetricType + Clone> Metric<T> {
         self.mmap_view = mmap_view;
     }
 
-    pub (super) fn write_val(&self, cursor: &mut Cursor<&mut [u8]>) -> io::Result<()> {
-        self.val.write_to_writer(cursor)
+    pub (super) fn write_val<W: Write>(&self, writer: &mut W) -> io::Result<()> {
+        self.val.write_to_writer(writer)
     }
 }
 
+pub struct InstanceMetric<T> {
+    pub (super) metrics: HashMap<String, Metric<T>>,
+    sem: Semantics,
+    unit: Unit,
+    indom: u32,
+    shorthelp: String,
+    longhelp: String,
+}
+
+impl<T: MetricType + Clone> InstanceMetric<T> {
+    pub fn new(
+        sem: Semantics,
```

Comment:
An InstanceMetric does have its own name, and values are represented as `name[instance1]`, `name[instance2]` etc.